syntax = "proto3";

package loan;

service LoanService {
  rpc GetLoan (GetLoanRequest) returns (LoanResponse) {}
  rpc UpdateLoanStatus (UpdateLoanStatusRequest) returns (LoanResponse) {}
  rpc UpdatePayment (PaymentUpdateRequest) returns (PaymentUpdateResponse) {}
  rpc GenerateCheckoutSession (CheckoutSessionRequest) returns (CheckoutSessionResponse) {}
}

message GetLoanRequest {
  int32 loan_id = 1;
}

message UpdateLoanStatusRequest {
  int32 loan_id = 1;
  string new_status = 2;
}

message LoanResponse {
  Loan loan = 1;
}

message Loan {
  int32 id = 1;
  int32 user_id = 2;
  int64 loan_amount_cents = 3;
  int32 loan_term_months = 4;
  int32 interest_rate = 5;
  string status = 6;
  int32 merchant_id = 7;
  string created_at = 8;
  string updated_at = 9;
}

message PaymentUpdateRequest {
  int32 loan_id = 1;
  string payment_status = 2;
}

message PaymentUpdateResponse {
  bool success = 1;
}

message CheckoutSessionRequest {
  int32 user_id = 1;
  int64 loan_amount_cents = 2;
  int32 merchant_id = 5;
  string order_id = 6;
  string success_redirect_url = 7;
  string cancel_redirect_url = 8;
}

message CheckoutSessionResponse {
  string checkout_url = 1;
}
